modules/porous_flow/include/materials/PorousFlowSingleComponentFluid.h

```diff
@@ -30,41 +30,50 @@ class PorousFlowSingleComponentFluid : public PorousFlowFluidPropertiesBase
   virtual void initQpStatefulProperties() override;
   virtual void computeQpProperties() override;
 
+  /// If true, this Material will compute density and viscosity, and their derivatives
+  const bool _compute_rho_nu;
+
+  /// If true, this Material will compute internal energy and its derivatives
+  const bool _compute_internal_energy;
+
+  /// If true, this Material will compute enthalpy and its derivatives
+  const bool _compute_enthalpy;
+
```

Comment:
I like this, but do we always need internal energy and enthalpy together? Could we simplify it to an `isothermal = true/false` and turn them both on/off with that?